### PR TITLE
Add AVX2.0 implementation of ProjectionProfile function

### PR DESCRIPTION
**Pull request by 0x72D0**
PR for #116

the performance when horizontal is false:

screenshot_20181108_133057

the performance when horizontal is true:

screenshot_20181108_132833

all units tests pass.

I also fix a bug with the random value, when I was doing the units tests for projection profile, the randomization of the horizontal variable always gave me true.

### CODE REVIEW

**Comment by 0x72D0** [comment]
ok I revert it :)

File: test/performance_tests/performance_test_image_function.cpp [hunk 1]
```diff
@@ -327,6 +335,7 @@ namespace image_function
     SET_FUNCTION( LookupTable        )
     SET_FUNCTION( Maximum            )
     SET_FUNCTION( Minimum            )
```
Comment (ihhub) [inline test/performance_tests/performance_test_image_function.cpp]: Please a test for function_pool as well as it uses your new code internally. It should work even faster. Additionally, please align **)** symbol with other SET_FUNCTION lines by white space shift :)

**Comment by 0x72D0** [comment]
I don't know why but it's not faster with function_pool:

screenshot_20181108_224505

**Comment by ihhub** [comment]
> I don't know why but it's not faster with function_pool:
> 
> screenshot_20181108_224505

Could you please verify that your new AVX code is called via function pool? It might be some missing hookups but logically adding one line for function table (which you did) fixes the problem.

**Comment by 0x72D0** [comment]
also I forgot to warn you about this code:

https://github.com/0x72D0/penguinV/blob/f01e1897a2ccef26d404933e4b52463923ac92c2/test/performance_tests/performance_test_image_function.cpp#L228

since I don't know how to test the true and false horizontal at the same time, I fixed it to false

**Comment by ihhub** [comment]
@0x72D0 thanks a lot for your contribution to the library!

**Pull request merged**